components/projects/EnvironmentVariableEditor.tsx

```diff
@@ -0,0 +1,192 @@
+            >
+              <CustomInput
+                name={envPair.envKey}
+                onFocus={() => (pastingInputIndex.current = index)}
+                type="text"
+                defaultValue={envPair.envKey}
+                className="my-1 w-full font-mono"
+                onPaste={handlePaste}
+                placeholder="eg. CLIENT_ID"
+              />
+
+              <div className="my-1 flex w-full items-center space-x-2">
+                <InputGroup
+                  full
+                  icon={<EyeIcon className="h-4 w-4" />}
+                  name={envPair.envValue}
+                  type="password"
+                  autoComplete="off"
+                  defaultValue={envPair.envValue}
+                  className="font-mono"
+                />
+
+                <Button
+                  onClick={() => handleRemoveEnvPairClick(index)}
+                  className="rounded"
+                  outline
+                  small
+                >
+                  <XIcon className="h-4 w-4" />
+                </Button>
+              </div>
+            </div>
+          ))}
+
+          <div className="mt-4 px-4">
+            <Button small outline onClick={() => handleAddMoreEnvClick()}>
+              Add more
+            </Button>
+          </div>
+        </div>
+      ) : (
+        <div {...getRootProps()} className="py-16 text-center">
+          <DragDropIcon className="mx-auto h-12 w-12" />
+          <h3 className="mt-2 text-xl text-gray-400">
+            Drag and drop .env files
+          </h3>
+          <input
+            {...getInputProps()}
+            type="file"
+            className="hidden"
+            accept="env"
+          />
+          <p className="mx-auto mt-1 max-w-md text-sm text-gray-200 text-lighter">
+            You can also{" "}
+            <span
+              onClick={open}
+              className="text-teal-300 transition duration-300 hover:cursor-pointer hover:underline"
+            >
+              click here
+            </span>{" "}
+            to import, copy/paste contents in .env file, or create{" "}
+            <span
+              onClick={handleAddMoreEnvClick}
+              className="text-teal-300 transition duration-300 hover:cursor-pointer hover:underline"
+            >
+              one at a time.
+            </span>
+          </p>
+        </div>
+      )}
+    </Container>
+  );
+}
+
+function CustomInput({
+  disabled,
+  className,
+  ...props
+}: ComponentProps<"input">) {
+  return (
+    <input
+      {...props}
+      disabled={disabled}
+      className={`${
+        disabled ? "cursor-not-allowed" : ""
+      } ${className} block appearance-none rounded border border-light/50 bg-darker px-3 py-2 placeholder-light shadow-sm ring-1 ring-light/50 focus:border-dark focus:outline-none focus:ring-light sm:text-sm`}
+    />
+  );
+}
+
+function DragDropIcon({ className } : { className?: string}) {
```

Comment:
Please take this to `components/icons` directory

Reply:
taken out